docker-compose.override.yml

```diff
@@ -41,24 +38,39 @@ services:
       done &&
       exec $$0 \"$$@\""
     command: minio server /data --address :5000 --console-address :5001
-    volumes:
-      # Any buckets used for testing provider data imported from s3 should be subdirectories under 
-      # /tests/s3-data/
-      # The trick to the issues here (https://wordpress.slack.com/archives/C02012JB00N/p1655118749223179)
-      # was to have a single volume, and to represent each bucket as a subdirectory under that.
-      # BUT, the test runs still can't see the inaturalist-open-data bucket, even though I have 
-      # verified through docker that the files exist on the container.
-      # TO DO: The openverse user has root access to all buckets on this minio instance, which
-      # makes sense for openverse-owned buckets, but not for read-only buckets owned by providers.
-      # Maybe we add a minio/mc + minio/minio pair with more realistic permissions for buckets
-      # containing sample provider data?
-      - ./tests/s3-data:/data:rw
     healthcheck:
       test: ["CMD", "curl", "-f", "http://localhost:5010/minio/health/live"]
       interval: 30s
       timeout: 20s
       retries: 3
 
+  load_to_s3:
+    image: minio/mc:latest
+    env_file:
+      - .env
+    depends_on:
+      - s3
+    volumes:
+      # Buckets for testing provider data imported from s3 are subdirectories under 
+      # /tests/s3-data/
+      - ./tests/s3-data:/data:rw
+    # Loop through subdirectories mounted to the volume and load them to s3/minio.
+    # This takes care of filesystem delays on some local dev environments that may make
+    # minio miss files included directly in the minio volume.
+    # More info here: https://stackoverflow.com/questions/72867045
+    # This does *not* allow for testing permissions issues that may come up in real AWS.
+    # And, if you remove files from /tests/s3-data, you will need to use `just down -v`
+    # and `just up` or `just recreate` to see the minio bucket without those files.
+    entrypoint: >
+      /bin/sh -c "
+      /usr/bin/mc config host add s3 http://s3:5000 ${AWS_ACCESS_KEY} ${AWS_SECRET_KEY};
+      cd /data;
+      for b in */ ; do
```

Comment:
Question! This seems a lot nicer than having to manually list the buckets in the environment variable like we were before, but in a fresh pull are those directories present? It seems like they're actually git-ignored, so for a new pull these buckets wouldn't actually get put in?

If that's the case I think we'll need to keep the environment variable approach (or just inline the variable into this script, which is probably nicer as it doesn't pollute the environment of the running container).

Reply:
The directories here aren't empty and shouldn't be gitignore'd, because they have the small test files that `test_inaturalist.py` relies on. They're similar to the resources sub-directories under tests\dags\providers\provider_api_scripts. But, because we're testing the database pulling directly from S3, it seemed important to have them in minio. Does that make sense? Is it ok that they're zipped binaries? That's what we'll get from inaturalist S3.

Reply:
Yes, that's fine for the new inaturalist ones, I was talking about the _other_ buckets, the ones that were already being created by the old `entrypoint` script. Though I suppose the rest of the tests are still passing so it must not actually be a problem :thinking: